capstone-ui/src/components/Feed/Feed.jsx

```diff
@@ -4,33 +4,49 @@ import "./Feed.css";
 import axios from "axios";
 import LoadingSpinner from "../LoadingSpinner/LoadingSpinner";
 import Recommendations from "../Recommendations/Recommendations";
-import { catchErrors } from "../../utils";
+import { catchErrors, showPopup, hidePopup } from "../../utils";
 import { getRecommendedUsers } from "../../recommendationUtils";
+import ProfileDetails from "../ProfileDetails/ProfileDetails";
 
 export default function Feed({ username, profile, token }) {
   const [feed, setFeed] = useState(null);
   const [recommendations, setRecommendations] = useState(null);
   const [shouldUpdateFeed, setShouldUpdateFeed] = useState(false);
+  const [isHovering, setIsHovering] = useState(false);
+  const [hoverUsername, setHoverUsername] = useState(null);
+  const [shouldUpdateProfileDetails, setShouldUpdateProfileDetails] =
+    useState(false);
+
+  const handleMouseOver = (username) => {
+    setIsHovering(true);
+    showPopup();
+    setHoverUsername(username);
+    setShouldUpdateProfileDetails(true);
+  };
+
+  const handleMouseOut = () => {
+    setIsHovering(false);
+    hidePopup();
+    setShouldUpdateProfileDetails(false);
+  };
 
   React.useEffect(() => {
     const fetchData = async () => {
-      const response = await axios.get(
+      const feedResult = await axios.get(
         `http://localhost:3001/user/feed/${username}`
       );
-      setFeed(response.data);
+      setFeed(feedResult.data);
 
       let genresResult = await axios.get(
         `http://localhost:3001/user/topgenres/${username}`
       );
       let artistsResult = await axios.get(
         `http://localhost:3001/user/topartists/${username}`
       );
```

Comment:
I feel like this is likely called elsewhere and prob can extract to a util function? Don't need to do for this PR but something to keep in mind.

Reply:
Good point I have a lot of the same calls in many of my useEffects. I will refactor these when everything been merged so it's more organized and I can keep track